test/runtests.jl

```diff
@@ -107,3 +145,36 @@ end
         rmprocs([left, right])
     end
 end
+
+@testset "@defer" begin
+    channel = eval(macroexpand(quote
```

Comment:
Both tests with `@defer` should exist as well as tests with `macroexpand`. Neither should include `eval`. Just test that the returned code is correct and that no errors are thrown when running `macroexpand`. Also test for error conditions, which would have uncovered your "`AssertertionError`".